src/main/java/org/terasology/projsndwv/ui/ApiaryScreen.java

```diff
@@ -88,4 +100,32 @@ public EntityRef get() {
             return entityRef;
         }
     }
+
+    @Override
+    public void update(float delta) {
```

Comment:
Ideally, this function shouldn't exist. You can use a binding to update the bar, and either a binding or methods that are triggered when needed for the rest

Reply:
Update is necessary for mating. I figured not using it for lifespan as well was unnecessary optimization.